features/TESTS/filesystem/heap_block_device/main.cpp

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+#include "HeapBlockDevice.h"
+#include <stdlib.h>
+
+using namespace utest::v1;
+
+/* It is not possible to build a KL25Z image with IAR including the file system if
+ * stack tracking statistics are enabled. If this is the case, build dummy
+ * tests.
+ */
+#if ! defined(TOOLCHAIN_IAR) && ! defined(TARGET_KL25Z) && ! defined(MBED_STACK_STATS_ENABLED)
```

Comment:
Is this just KL25Z specific or because target does not have enough memory (this might lead to skip this test for many targets, the list will grow here) ? 

I cant spot here a reason why it fails to build.